File: frontends/main/src/app/configurator/services/api/validator.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs';
import {Validator} from '@core/model/validator';

@Injectable({
	providedIn: 'root'
})
export class ValidatorService {
	constructor(private http: HttpClient) {}

	getValidators(projectId: string): Observable<Validator[]> {
		return this.http.get<Validator[]>(`/api/superuser/configurator/projects/${projectId}/config/validators`);
	}

	getValidator(projectId: string, validatorId: string): Observable<Validator> {
		return this.http.get<Validator>(`/api/superuser/configurator/projects/${projectId}/config/validators/${validatorId}`);
	}

	createValidator(projectId: string, validator: Validator): Observable<Validator> {
		return this.http.post<Validator>(`/api/superuser/configurator/projects/${projectId}/config/validators`, validator);
	}

	updateValidator(projectId: string, validatorId: string, validator: Validator): Observable<Validator> {
		return this.http.put<Validator>(`/api/superuser/configurator/projects/${projectId}/config/validators/${validatorId}`, validator);
	}

	deleteValidator(projectId: string, validatorId: string): Observable<void> {
		return this.http.delete<void>(`/api/superuser/configurator/projects/${projectId}/config/validators/${validatorId}`);
	}
}

File: backend/src/main/java/ch/rodano/core/services/bll/configurator/ValidatorService.java
package ch.rodano.core.services.bll.configurator;

import java.util.List;
import java.util.UUID;

import ch.rodano.api.config.ValidatorDTO;

public interface ValidatorService {

	List<ValidatorDTO> getValidators(UUID projectId, String view);

	default List<ValidatorDTO> getValidators(final UUID projectId) {
		return getValidators(projectId, "summary");
	}

	ValidatorDTO getValidator(UUID projectId, UUID validatorId);

	ValidatorDTO createValidator(UUID projectId, ValidatorDTO validator);

	ValidatorDTO updateValidator(UUID projectId, UUID validatorId, ValidatorDTO validator);

	void deleteValidator(UUID projectId, UUID validatorId);
}
